samples/MvcSubAreaSample.Web/SubAreaViewLocationExpander.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.Mvc.Razor;
+
+namespace MvcSubAreaSample.Web
+{
+    public class SubAreaViewLocationExpander : IViewLocationExpander
+    {
+        private const string _subAreaKey = "subarea";
+
+        public IEnumerable<string> ExpandViewLocations(ViewLocationExpanderContext context,
+            IEnumerable<string> viewLocations)
+        {
+            if (context.Values.ContainsKey(_subAreaKey))
+            {
+                var subArea = context.Values[_subAreaKey];
```

Comment:
I guess this would be fine (since it's simply pulling the value out of the ViewLocationExpander dictionary. `PopulateValues` would need to change to use the GetNormalizedRouteValue method.